src/legacy/core_plugins/input_control_vis/public/control/filter_manager/phrase_filter_manager.ts

```diff
@@ -18,37 +18,37 @@
  */
 
 import _ from 'lodash';
-import { FilterManager } from './filter_manager.js';
+
+import { PhraseFilter } from 'src/plugins/data/common/es_query/filters';
+import { IndexPattern } from '../../legacy_imports';
+import { FilterManager } from './filter_manager';
 import {
+  FilterManager as QueryFilterManager,
   esFilters,
+  IFieldType,
 } from '../../../../../../plugins/data/public';
 
 export class PhraseFilterManager extends FilterManager {
-  constructor(controlId, fieldName, indexPattern, queryFilter) {
+  constructor(
+    controlId: string,
+    fieldName: string,
+    indexPattern: IndexPattern,
+    queryFilter: QueryFilterManager
+  ) {
     super(controlId, fieldName, indexPattern, queryFilter);
   }
 
-  /**
-   * Convert phrases into filter
-   *
-   * @param {array} phrases
-   * @return {object} query filter
-   *   single phrase: match query
-   *   multiple phrases: bool query with should containing list of match_phrase queries
-   */
-  createFilter(phrases) {
-    let newFilter;
+  createFilter(phrases: any): PhraseFilter {
+    let newFilter: PhraseFilter;
+    // TODO: Fix type to be required
+    const value = this.indexPattern.fields.getByName(this.fieldName) as IFieldType;
+
     if (phrases.length === 1) {
```

Comment:
Can't you use `esFilters.buildFilter` to replace at least some of this logic?

Reply:
see above comment